src/NUnitFramework/framework/Internal/Tests/Test.cs

```diff
@@ -92,9 +92,9 @@ private Test(string? pathName, string name, ITypeInfo? typeInfo, IMethodInfo? me
             TearDownMethods = new IMethodInfo[0];
         }
 
-        private static string GetNextId()
+        internal string SetNextId()
         {
-            return IdPrefix + unchecked(_nextID++);
+            return Id = IdPrefix + unchecked(_nextID++);
```

Comment:
I'm uncomfortable with mixing getting and setting. I'd prefer if you added `SetNextId` and left `GetNextId`.

Reply:
Re-added GetNextId